ground/src/main/java/com/google/android/ground/PermissionDeniedDialog.kt

```diff
@@ -0,0 +1,122 @@
+import androidx.compose.material3.AlertDialog
+import androidx.compose.material3.Button
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedButton
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalUriHandler
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.google.android.ground.ui.compose.HyperlinkText
+import com.google.android.ground.ui.theme.AppTheme
+import timber.log.Timber
+
+@Composable
+fun PermissionDeniedDialog(signupLink: String?, onSignOut: () -> Unit, onCloseApp: () -> Unit) {
+  AlertDialog(
+    onDismissRequest = {},
+    title = {
+      Row(verticalAlignment = Alignment.CenterVertically) {
+        Icon(
+          painter = painterResource(id = R.drawable.baseline_warning_24),
+          contentDescription = "Alert Icon",
+          tint = MaterialTheme.colorScheme.error,
+        )
+
+        // Add some space between icon and title
+        Spacer(modifier = Modifier.width(12.dp))
+
+        Text(
+          stringResource(id = R.string.permission_denied),
+          style = MaterialTheme.typography.titleLarge,
+        )
+      }
+    },
+    text = {
+      Column {
+        if (signupLink.isNullOrEmpty()) {
+          Text(
+            stringResource(R.string.admin_request_access),
+            style = MaterialTheme.typography.bodyMedium,
+          )
+        } else {
+          SignupLink(signupLink)
+        }
+
+        // Empty line
+        Text(text = "")
+
+        Text(
+          stringResource(R.string.signout_warning),
+          style = MaterialTheme.typography.bodyMedium,
+          color = MaterialTheme.colorScheme.error,
+        )
+      }
+    },
+    dismissButton = {
+      OutlinedButton(onClick = { onSignOut() }) { Text(stringResource(id = R.string.sign_out)) }
+    },
+    confirmButton = {
+      Button(onClick = { onCloseApp() }) { Text(stringResource(id = R.string.close_app)) }
+    },
+  )
+}
+
+@Composable
+private fun SignupLink(signupLink: String) {
+  val uriHandler = LocalUriHandler.current
+
+  HyperlinkText(
+    textStyle = MaterialTheme.typography.bodyMedium,
+    fullTextResId = R.string.signup_request_access,
+    linkTextColor = MaterialTheme.colorScheme.primary,
+    hyperLinks =
+      mapOf(
+        "sign_up_link" to
+          {
+            runCatching { uriHandler.openUri(signupLink) }
+              .onFailure { Timber.e(it, "Failed to open sign-up link") }
+          }
+      ),
+  )
+}
+
+@Preview
+@Composable
+fun PermissionDeniedDialogWithSignupLinkPreview() {
+  AppTheme {
+    PermissionDeniedDialog(signupLink = "www.google.com", onSignOut = {}, onCloseApp = {})
```

Comment:
Can we declare this in a constant in BuildConfig as "", and only show the version with the link if the constant `isNotEmpty()`? That was the Open Foris fork will just need to modify the constant value.

Reply:
Done